src/Codeception/Module/Db.php

```diff
@@ -949,6 +949,68 @@ public function grabFromDatabase(string $table, string $column, array $criteria
         return $this->proceedSeeInDatabase($table, $column, $criteria);
     }
 
+    /**
+     * Fetches a whole entry from a database.
+     * Provide table name, desired column and criteria.
+     *
+     * ``` php
+     * <?php
+     * $mail = $I->grabEntryFromDatabase('users', array('name' => 'Davert'));
+     * ```
+     * Comparison expressions can be used as well:
+     *
+     * ```php
+     * <?php
+     * $post = $I->grabEntryFromDatabase('posts', ['num_comments >=' => 100]);
+     * $user = $I->grabEntryFromDatabase('users', ['email like' => 'miles%']);
+     * ```
+     *
+     * Supported operators: `<`, `>`, `>=`, `<=`, `!=`, `like`.
+     *
+     * @return mixed Returns a single entry value or false if it fails
```

Comment:
`@return array<string, mixed>|false Returns a single entry value or false if there is no matching row.`
`PDO::ATTR_ERRMODE` is set to `PDO::ERRMODE_EXCEPTION`, so there won't be `false` if query fails.

Wouldn't it be a good idea to throw AssertionError if there is no matching row?

Reply:
> Wouldn't it be a good idea to throw AssertionError if there is no matching row?
I'm usually not a big fan on triggering behaviour on behalf of the user, but in this particular case I would agree since the user should use another method if he was to test the non-existence of a row.